### ISSUE TITLE
log all actions (including movements, airlifts, disbands, buying)

### ISSUE THREAD

**Issue by hugomflavio** [longturn/freeciv21#2395]
**Is your feature request related to a problem? Please describe.**
Admins should be able to retrace every single action performed as the game goes on. That is currently not the case.

**Describe the solution you'd like**
Log every single action. Require e.g. a global observer with hack command level to see those detailed logs.

**Describe alternatives you've considered**
Dump every single action into a file/repository that only the game admin has access to.

**Additional context**
This will help the admin make informed decisions when e.g. a rule is reported to have been broken.

**Comment by jwrober** [comment]
I see this as broken down into many steps:

- [ ] Do the thing directly asked, plus... 
- [ ] New hack level - game admin
- [ ] New "game admin" role that we can place in the players.serv file. Should probably support a 1 to many relationship so more than one person can be listed as a game admin
- [ ] Change the server to accept the game admin role
- [ ] Change the client to allow a game admin to connect as a global observer (all seeing) to the server with the role
- [ ] Change the client to give a new interface that allows the game admin to see what happened
- [ ] Then #2397 

We also have the problem of #1762 that probably should be resolved if we do this. That's a lot of game data that is passed over the internet in plain text.

**Comment by lmoureaux** [comment]
> New hack level - game admin

I'd be more in favor of a "permissions" API for fcdb that queries whether a user can do some action (or, since we'll have to send this to the client, the list of actions a user can do).

> Change the client to allow a game admin to connect as a global observer (all seeing) to the server with the role

It is already possible for game admins to `/observe`. This is handled through fcdb.

**Comment by hugomflavio** [comment]
Very busy days for me nowadays with the new job, but once things calm down I'll happily help out where I can.

**Comment by XHawk87** [comment]
A lot of the logging can already be done through the Lua API. Certainly all movement and all unit actions. The only one you mention that it can't log right now is when a city buys something since that's not a unit action. Though adding more Lua events is definitely something I'm in favour of. 

It can be sent to the server log for a server admin to pass on to a game admin as needed, or sent via notification to a player. Perhaps a special `Game Admin LT##` account.

**Comment by XHawk87** [comment]
> That account would need to play.

I don't think it actually does. We can create a nation for them after the map gen using Lua before the logging begins, it doesn't even need any units or cities to be able to `/take` it and view the events.
